### PR TITLE
[dagster-sigma] Use aiohttp to fetch data from Sigma APIs

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Rewrites the internals of `dagster-sigma` to fetch from APIs using aiohttp instead of requests, letting us parallelize many fetches.

Drops the owner fetch for now, since this can also introduce delay and is not super important.

## Test Plan

Existing unit tests, tested vs live instance.

## Changelog

> [dagster-sigma] The Sigma integration now fetches initial API responses in parallel, speeding up initial load.

### CODE REVIEW

**Comment by benpankow** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25374?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25374** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25374?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25371** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25371?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-sigma/dagster_sigma/resource.py [hunk 1]
```diff
@@ -86,197 +88,257 @@ def api_token(self) -> str:
...
+                    **SIGMA_PARTNER_ID_TAG,
+                },
+            ) as response:
+                response.raise_for_status()
+                return await response.json()
 
     @cached_method
-    def _fetch_workbooks(self) -> List[Dict[str, Any]]:
-        return self._fetch_json("workbooks")["entries"]
+    async def _fetch_workbooks(self) -> List[Dict[str, Any]]:
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-sigma/dagster_sigma/resource.py:112]: I think using cached methods with async might cause some problems, but you're probably avoiding them by returning the awaited thing so it's probably fine -- do any codepaths currently invoke any of these methods more than once?

**Review by OwenKephart — APPROVED** [review]
I _think_ the cached method async thing will be fine, but just would be good to double check

File: python_modules/libraries/dagster-sigma/dagster_sigma/resource.py (hunk 1 above)
Comment (benpankow) [reply python_modules/libraries/dagster-sigma/dagster_sigma/resource.py:112]: https://github.com/dagster-io/dagster/pull/25371 😄

**Pull request merged**